Tabloid/Controllers/TagController.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+using Tabloid.Repositories;
+using Microsoft.VisualBasic;
+using System.Security.Claims;
+using Tabloid.Models.ViewModels;
+using Microsoft.AspNetCore.Authorization;
+using System;
+using Tabloid.Models;
+
+namespace Tabloid.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class TagController : ControllerBase
+    {
+        private readonly ITagRepository _tagRepository;
+
+        public TagController(ITagRepository tagRepository)
+        {
+            _tagRepository = tagRepository;
+        }
+
+        // GET: TagController
+        [HttpGet("Index")]
+        public ActionResult Index()
+        {
+            var tags = _tagRepository.GetAllTags();
+            return Ok(_tagRepository.GetAllTags());
+        }
+
+        // POST: api/Tag
+        [Authorize(Roles = "Admin")]
```

Comment:
This won't work for the API code the way it worked in MVC. Take a look at an example from Streamish to see how to do this on the server side.